### PR TITLE
Improve the Admin Console loading options

### PR DESCRIPTION
**Pull request by avpinchuk**
Improve the Administration Console loading options:

* **ALWAYS** loads the Console at server startup (eager loading)
* **DEFAULT** loads the Console only after first access to it (on demand loading, used by default)
* Implements **NEVER** loading option that disable the Admin Console.

Also cleaned up the console-related code.

### Synopsis

```sh
asadmin set server.admin-service.property.adminConsoleStartup={default|always|never}
```

### Example

This enforces Admin Console to be loaded at server startup:

```sh
asadmin set server.admin-service.property.adminConsoleStartup=always
```

After setting up the property the domain must be restarted to apply changes.

### CODE REVIEW

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/adapter/AdminConsoleAdapter.java [hunk 1]
```diff
@@ -162,165 +164,161 @@ public HttpHandler getHttpService() {
...
-                }
-                if (logger.isLoggable(Level.FINE)) {
-                    logger.log(Level.FINE,
-                            ioe.toString(),
-                            ioe);
-                }
+                handleResourceRequest(request, response);
+            } catch (IOException e) {
+                logger.log(Level.SEVERE, KernelLoggerInfo.consoleResourceError, new Object[] {request.getRequestURI(), e});
+                logger.log(Level.FINE, e, e::toString);
```
Comment (dmatej) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/adapter/AdminConsoleAdapter.java:209]: Hmm, I forgot this even exists: `public void log(Level level, Throwable thrown, Supplier<String> msgSupplier) {`

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/adapter/InstallerThread.java [hunk 2]
```diff
@@ -79,120 +87,127 @@ public void run() {
...
-                ApplicationRef aref = s.createChild(ApplicationRef.class);
-                aref.setRef(app.getName());
-                aref.setEnabled(Boolean.TRUE.toString());
-                aref.setVirtualServers(getVirtualServerList()); //TODO
-                arefs.add(aref);
-                return true;
+        logger.log(Level.FINE, "Installing the Admin Console Application...");
+
+        // Create the application entry in domain.xml
+        ConfigCode code = proxies -> {
```
Comment (dmatej) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/adapter/InstallerThread.java:115]: Quite big anonymous class, maybe it should have a name and own file.

**Review by dmatej — APPROVED** [review]

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/adapter/InstallerThread.java (hunk 2 above)
Comment (avpinchuk) [reply nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/adapter/InstallerThread.java:115]: Thanks for advice. Please DO NOT MERGE. I'll change it later.

Comment (avpinchuk) [reply nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/adapter/InstallerThread.java]: Done.

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/adapter/LocalStrings.properties [hunk 3]
```diff
@@ -54,12 +41,11 @@ state.prepareRedeploy=Preparing to upgrade Admin Console Application...
 state.cleanupFailed=Exception while cleaning previous instance of admin console
 state.backupFailed=Cannot rename __admingui to __admingui.backup
 state.cleaningBackup=Cleaning up temporary backup file...
-state.cleanupFailed=Temporary backup file removed
+state.backupCleaned=Temporary backup file removed
```
Comment (pzygielo) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/adapter/LocalStrings.properties:44]: nice finding

**Review by dmatej — APPROVED** [review]

**Review by arjantijms — APPROVED** [review]

**Comment by OndroMih** [comment]
@avpinchuk, this is a nice addition. Wiuld be good to document how to use it (set the loading option via asadmin or admin console) at least here in the PR description. Ideally also in one of the GlassFish guides somewhere.

**Comment by avpinchuk** [comment]
@OndroMih, I added a description to this PR. 

Also I'll think about which reference guide to change.

**Pull request merged**